src/servers/src/http/influxdb_result_v1.rs

```diff
@@ -0,0 +1,242 @@
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use common_error::ext::ErrorExt;
+use common_query::Output;
+use common_recordbatch::{util, RecordBatch};
+use common_telemetry::{debug, error};
+use schemars::JsonSchema;
+use serde::{Deserialize, Serialize};
+use serde_json::Value;
+use snafu::ResultExt;
+
+use crate::error::{Error, ToJsonSnafu};
+use crate::http::Epoch;
+
+#[derive(Debug, Default, Serialize, Deserialize, JsonSchema)]
+pub struct SqlQuery {
+    pub db: Option<String>,
+    // Returns epoch timestamps with the specified precision.
+    // Both u and µ indicate microseconds.
+    // epoch = [ns,u,µ,ms,s],
+    pub epoch: Option<String>,
+    pub sql: Option<String>,
+}
+
+#[derive(Debug, Serialize, Deserialize, JsonSchema)]
+pub struct InfluxdbRecordsOutput {
+    // The SQL query does not return the table name, but in InfluxDB,
+    // we require the table name, so we set it to an empty string “”.
+    name: String,
+    pub(crate) columns: Vec<String>,
+    pub(crate) values: Vec<Vec<Value>>,
+}
+
+impl InfluxdbRecordsOutput {
+    pub fn new(columns: Vec<String>, values: Vec<Vec<Value>>) -> Self {
+        Self {
+            name: "".to_string(),
+            columns,
+            values,
+        }
+    }
+}
+
+impl TryFrom<(Option<Epoch>, Vec<RecordBatch>)> for InfluxdbRecordsOutput {
+    type Error = Error;
+
+    fn try_from(
+        (epoch, recordbatches): (Option<Epoch>, Vec<RecordBatch>),
+    ) -> Result<InfluxdbRecordsOutput, Self::Error> {
+        if recordbatches.is_empty() {
+            Ok(InfluxdbRecordsOutput::new(vec![], vec![]))
+        } else {
+            // Safety: ensured by previous empty check
+            let first = &recordbatches[0];
+            let columns = first
+                .schema
+                .column_schemas()
+                .iter()
+                .map(|cs| cs.name.clone())
+                .collect::<Vec<_>>();
+
+            let mut rows =
+                Vec::with_capacity(recordbatches.iter().map(|r| r.num_rows()).sum::<usize>());
+
+            for recordbatch in recordbatches {
+                for row in recordbatch.rows() {
+                    let value_row = row
+                        .into_iter()
+                        .map(|value| {
+                            if let Some(epoch) = epoch {
+                                if let datatypes::value::Value::Timestamp(ts) = &value {
+                                    if let Some(timestamp) = epoch.convert_timestamp(*ts) {
+                                        return Value::try_from(
+                                            datatypes::value::Value::Timestamp(timestamp),
+                                        );
+                                    }
+                                }
+                            }
+                            Value::try_from(value)
```

Comment:
To reduce a layer of deep nested "if"s:

```suggestion
                            let value = match (epoch, &value) {
                                (Some(epoch), datatypes::value::Value::Timestamp(ts)) =>
                                    if let Some(timestamp) = epoch.convert_timestamp(*ts) {
                                        datatypes::value::Value::Timestamp(timestamp),
                                    } else {
                                        value
                                    }
                                },
                                _ => value
                            }
                            Value::try_from(value)
```